src/core/public/chrome/global_search/global_search_service.ts

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { ReactNode } from 'react';
+
+export enum SearchObjectTypes {
+  PAGES = 'pages',
+  SAVED_OBJECTS = 'saved_objects',
+}
+/**
+ * @experimental
+ */
+export interface GlobalSearchStrategy {
+  /**
+   * unique id of this strategy
+   */
+  id: string;
+  /**
+   * search object type
+   * @type {SearchObjectTypes}
+   */
+  type: SearchObjectTypes;
+  /**
+   * do the search and return search result with a React element
+   * @param value search query
+   * @param callback callback function when search is done
+   */
+  doSearch(value: string, callback?: () => void): Promise<ReactNode[]>;
```

Comment:
How about `run`. This works well with the term command too. So it will be `command.run(props)`

Reply:
rename this method name to `run` as well